src/appleseed/foundation/math/ray.h

```diff
@@ -83,6 +83,11 @@ class Ray
     template <typename U>
     Ray(const Ray<U, N>& rhs);
 
+    ValueType get_length() const;
```

Comment:
Move after `is_finite()` and add short method description, in particular the fact that the ray must be finite.